### PR TITLE
Adding rad init command changes to support irsa

### PR DESCRIPTION
**Pull request by vishwahiremat**
# Description

- Added changes to `rad init --full` command to add types i.e accesskey,irsa while configuring aws provider
image
- And adding prompts to accept role arn
- refactoring the code with switch cases to handle it differently for both cases.
- updated the tests

## Type of change

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #issue_number

### CODE REVIEW

File: pkg/cli/aws/client.go [hunk 1]
```diff
@@ -58,13 +63,19 @@ func (c *client) GetCallerIdentity(ctx context.Context, region string, accessKey
...
-func (c *client) ListRegions(ctx context.Context, region string, accessKeyID string, secretAccessKey string) (*ec2.DescribeRegionsOutput, error) {
-	credentialsProvider := credentials.NewStaticCredentialsProvider(accessKeyID, secretAccessKey, "")
-	ec2Client := ec2.New(ec2.Options{
-		Region:      region,
-		Credentials: credentialsProvider,
-	})
-	result, err := ec2Client.DescribeRegions(ctx, &ec2.DescribeRegionsInput{})
+func (c *client) ListRegions(ctx context.Context, region string) (*ec2.DescribeRegionsOutput, error) {
+	// Load the AWS SDK config and credentials
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithSharedConfigProfile("default"))
```
Comment (nithyatsu) [inline pkg/cli/aws/client.go:68]: should "default" be a const?

File: pkg/cli/cmd/radinit/aws.go [hunk 2]
```diff
@@ -31,48 +31,94 @@ const (
 	QueryRegion = "us-east-1"
 
 	selectAWSRegionPrompt                 = "Select the region you would like to deploy AWS resources to:"
+	selectAwsCredentialKindPrompt         = "Select a credential kind for the AWS credential:"
```
Comment (nithyatsu) [inline pkg/cli/cmd/radinit/aws.go:34]: @willtsai @Reshrahim  cc-ing for reviewing all UX text.

File: pkg/cli/cmd/radinit/aws.go [hunk 3]
```diff
@@ -31,48 +31,94 @@ const (
...
+		return &aws.Provider{
+			AccountID:      accountId,
+			Region:         region,
+			CredentialKind: aws.AwsCredentialKindIRSA,
+			IRSA: &aws.IRSACredential{
+				RoleARN: roleARN,
+			},
+		}, nil
+	default:
+		return nil, clierrors.Message("Invalid Azure credential kind: %s", credentialKind)
```
Comment (nithyatsu) [inline pkg/cli/cmd/radinit/aws.go:116]: clierrors.Message("Invalid AWS credential kind: %s", credentialKind)

File: pkg/cli/cmd/radinit/aws.go [hunk 4]
```diff
@@ -31,48 +31,94 @@ const (
...
-		Region:          region,
-	}, nil
 }
 
-func (r *Runner) getAccountId(ctx context.Context, accessKeyID, secretAccessKey string) (string, error) {
-	callerIdentityOutput, err := r.awsClient.GetCallerIdentity(ctx, QueryRegion, accessKeyID, secretAccessKey)
+func (r *Runner) getAccountId(ctx context.Context) (string, error) {
+	callerIdentityOutput, err := r.awsClient.GetCallerIdentity(ctx, QueryRegion)
 	if err != nil {
 		return "", clierrors.MessageWithCause(err, "AWS credential verification failed.")
```
Comment (nithyatsu) [inline pkg/cli/cmd/radinit/aws.go:123]: Should this error message be enhanced to indicate we are using the .aws/credentials on cli side? for example 
AWS credential verification failed. Please use `aws configure` to set the credentials.  cc @Reshrahim @willtsai

Comment (Reshrahim) [reply pkg/cli/cmd/radinit/aws.go:123]: @vishwahiremat/ @nithyatsu  - does this verification occur immediately after the user inputs accesskey and secret ?

File: pkg/cli/cmd/radinit/aws.go (hunk 2 above)
Comment (Reshrahim) [reply pkg/cli/cmd/radinit/aws.go:34]: Synced with @vishwahiremat on the UX prompts.

File: pkg/cli/cmd/radinit/aws.go (hunk 4 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go:123]: yes thats right, we get the account id inforamtion after user inputs accesskey and secret

Comment (Reshrahim) [reply pkg/cli/cmd/radinit/aws.go:123]: if there is no AWS local config, can a user provide an account id and carry on the with flow?

Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go:123]: no, we still need AWS local config to get regions list. we expect the users to have aws configured locally.

Comment (Reshrahim) [reply pkg/cli/cmd/radinit/aws.go:123]: for both the cases something along the lines of, "AWS Cloud Provider setup failed, please use `aws configure` to set up the configuration. More information :https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html "

File: pkg/cli/aws/client.go (hunk 1 above)
Comment (vishwahiremat) [reply pkg/cli/aws/client.go]: not required anymore

File: pkg/cli/cmd/radinit/aws.go (hunk 2 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go:32]: added the fix as discussed

File: pkg/cli/cmd/radinit/aws.go (hunk 4 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go]: added the changes in this PR https://github.com/radius-project/radius/pull/7758

File: pkg/cli/aws/provider.go [hunk 5]
```diff
@@ -16,22 +16,44 @@ limitations under the License.
 
 package aws
 
+// AwsCredentialKind - Aws credential kinds supported.
+type AwsCredentialKind string
```
Comment (ytimocin) [inline pkg/cli/aws/provider.go:20]: What is the convention in our repository for `Aws`? Should it be `AWS` or do we use both `Aws` and `AWS`? I believe we should use `AWS`.

File: pkg/cli/aws/provider.go [hunk 6]
```diff
@@ -16,22 +16,44 @@ limitations under the License.
...
+
+	// AccessKey represents ucp credential kind for aws access key credentials.
+	AccessKey *AccessKeyCredential
+
+	// IRSA represents ucp credential kind for aws irsa credentials.
+	IRSA *IRSACredential
+}
+
+type AccessKeyCredential struct {
+	// AccessKeyID is the access key id for the AWS account.
```
Comment (ytimocin) [inline pkg/cli/aws/provider.go:49]: For example, we are using `AWS` here.

File: pkg/cli/azure/provider.go [hunk 7]
```diff
@@ -35,7 +35,7 @@ type Provider struct {
 	ServicePrincipal *ServicePrincipalCredential
 }
 
-// Wor specifies the properties of an Azure service principal
+// WorkloadIdentityCredential specifies the properties of an Azure service principal
```
Comment (ytimocin) [inline pkg/cli/azure/provider.go:38]: 🤔

File: pkg/cli/cmd/radinit/aws.go [hunk 8]
```diff
@@ -29,7 +29,10 @@ import (
 
 const (
 	selectAWSRegionPrompt                 = "Select the region you would like to deploy AWS resources to:"
+	selectAwsCredentialKindPrompt         = "Select a credential kind for the AWS credential:"
```
Comment (ytimocin) [inline pkg/cli/cmd/radinit/aws.go:32]: We use `AWS` and `Aws` in two lines that follow each other.

File: pkg/cli/cmd/radinit/aws.go [hunk 9]
```diff
@@ -29,7 +29,10 @@ import (
...
 	enterAWSIAMAcessKeyIDPrompt           = "Enter the IAM access key id:"
+	enterAWSRoleARNPrompt                 = "Enter the role ARN:"
+	enterAWSRoleARNPlaceholder            = "Enter IAM role ARN..."
```
Comment (ytimocin) [inline pkg/cli/cmd/radinit/aws.go:35]: Here you added `AWS` instead of `Aws`.

File: pkg/cli/cmd/radinit/aws.go [hunk 10]
```diff
@@ -39,37 +42,80 @@ const (
...
+		if err != nil {
+			return nil, err
+		}
+
+		region, err := r.selectAWSRegion(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		// Set the value for the Helm chart
```
Comment (ytimocin) [inline pkg/cli/cmd/radinit/aws.go:106]: We don't need these comments in production, I guess.

File: pkg/cli/cmd/radinit/aws.go [hunk 11]
```diff
@@ -123,3 +169,19 @@ func (r *Runner) buildAWSRegionsList(listRegionsOutput *ec2.DescribeRegionsOutpu
...
+func (r *Runner) selectAwsCredentialKind() (string, error) {
+	credentialKinds, err := r.buildAwsCredentialKind()
+	if err != nil {
+		return "", err
+	}
+
+	return r.Prompter.GetListInput(credentialKinds, selectAwsCredentialKindPrompt)
+}
+
+func (r *Runner) buildAwsCredentialKind() ([]string, error) {
```
Comment (ytimocin) [inline pkg/cli/cmd/radinit/aws.go:182]: This returns a list of available AWS credential kinds. Should we change the name? Also, can this ever return an error?

File: pkg/cli/cmd/radinit/environment.go [hunk 12]
```diff
@@ -108,8 +108,11 @@ func (r *Runner) CreateEnvironment(ctx context.Context) error {
 	}
 
 	if r.Options.CloudProviders.AWS != nil {
-		credential := r.getAWSCredential()
-		err := credentialClient.PutAWS(ctx, credential)
+		credential, err := r.getAWSCredential()
+		if err != nil {
+			return clierrors.MessageWithCause(err, "Failed to configure AWS credentials.")
+		}
+		err = credentialClient.PutAWS(ctx, credential)
```
Comment (ytimocin) [inline pkg/cli/cmd/radinit/environment.go:115]: nit: extra line before

File: pkg/cli/cmd/radinit/aws.go (hunk 11 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go]: updated it to remove error from the return type. 
and for the name today we follow the similar format in azure provider:
https://github.com/radius-project/radius/blob/b7484430568763f05fba769b325e6f0471092fcb/pkg/cli/cmd/radinit/azure.go#L307

File: pkg/cli/cmd/radinit/aws.go (hunk 10 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go:106]: we are doing it in azure as well, if its not recommended we can remove these comments

https://github.com/vishwahiremat/radius/blob/b7484430568763f05fba769b325e6f0471092fcb/pkg/cli/cmd/radinit/azure.go#L127-L128

File: pkg/cli/cmd/radinit/aws.go (hunk 9 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go:35]: I see we are using mix of these 2 formats, mostly i see `Aws` in ucp code and `AWS` on the cli side. I have updated it to `AWS` format

File: pkg/cli/aws/provider.go (hunk 5 above)
Comment (vishwahiremat) [reply pkg/cli/aws/provider.go]: I see we are using mix of these 2 formats, mostly i see Aws in ucp code and AWS on the cli side. I have updated it to AWS format

File: pkg/cli/aws/provider.go (hunk 6 above)
Comment (vishwahiremat) [reply pkg/cli/aws/provider.go:49]: updated it

File: pkg/cli/azure/provider.go (hunk 7 above)
Comment (vishwahiremat) [reply pkg/cli/azure/provider.go:38]: fixed a typo

File: pkg/cli/cmd/radinit/aws.go (hunk 8 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go]: updated it

File: pkg/cli/cmd/radinit/aws.go (hunk 9 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go:35]: updated it

File: pkg/cli/cmd/radinit/aws.go (hunk 10 above)
Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go:106]: today i see comments in azure provider section as well, we can remove it if its not recommended https://github.com/vishwahiremat/radius/blob/b7484430568763f05fba769b325e6f0471092fcb/pkg/cli/cmd/radinit/azure.go#L127-L128

File: pkg/cli/cmd/radinit/aws.go [hunk 13]
```diff
@@ -123,3 +169,15 @@ func (r *Runner) buildAWSRegionsList(listRegionsOutput *ec2.DescribeRegionsOutpu
 
 	return regions
 }
+
+func (r *Runner) selectAWSCredentialKind() (string, error) {
+	credentialKinds := r.buildAWSCredentialKind()
+	return r.Prompter.GetListInput(credentialKinds, selectAWSCredentialKindPrompt)
+}
+
+func (r *Runner) buildAWSCredentialKind() []string {
```
Comment (ytimocin) [inline pkg/cli/cmd/radinit/aws.go:178]: I guess we can also change the name of the function. Are we building something here? Is this the convention we are using? Isn't it more like `awsCredentialKinds`?

Comment (vishwahiremat) [reply pkg/cli/cmd/radinit/aws.go:178]: this is format we used in azure, we are using the similar format in azure provider.

**Review by ytimocin — APPROVED** [review]

**Pull request merged**